src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateClone, { ...action.extraData });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete stateClone[key];
+      }
+    }
+
+    if (action.type === 'clear') {
+      for (const key in stateClone) {
+        delete stateClone[key];
+      }
+    }
+
+    result.push(stateClone);
```

Comment:
Pushing the 'stateClone' to the 'result' array should be done outside of the 'if' or 'switch' block, after the current action is processed, to avoid repeating this operation in each case.